ERCS/erc-7655.md

```diff
@@ -0,0 +1,113 @@
+  /// @param operation The operation to derive a receipt for.
+  /// @param transactionLogs The logs of a transaction that executes the operation.
+  /// @return status Whether the operation succeeded or failed.
+  /// @return operationLogs The logs emitted specifically due to this operation.
+  /// @return gasUsed The gas used specifically due to this operation.
+  function operationReceipt(
+    Endorser.Operation calldata operation,
+    Log[] calldata transactionLogs
+  ) external view returns (
+    Status status,
+    Log[] memory operationLogs,
+    uint256 gasUsed
+  );
+
+}
+```
+
+When `operationFilter` is called with an operation that the endorser recognizes, the endorser SHOULD return an origin address and an array of possible topics for that position in a matching log's topics, as described in the JSON-RPC API documentation for the `eth_getLogs` and `eth_newFilter` methods, with the exception that an empty array in a position behaves equivalently to the `null` value.
+
+When `operationReceipt` is called with an operation that the endorser recognizes and the logs of the transaction where the operation was executed, the endorser SHOULD return the success status of the operation, a subsequence of logs that were emitted during the execution of the operation, and the amount of gas used during execution of the operation.
+
+## Rationale
+
+`operationFilter` addresses the first requirement for detecting when an operation is mined without any knowledge of the bundler that executed the operation on-chain.
+Users can use the result of this function to issue an `eth_getLogs` request to a JSON-RPC node to find potential transactions where the operation was executed.
+
+`operationReceipt` addresses the other requirements of determining if the operation succeeded, identifying which logs of the encompassing transaction were executed in the context of the operation in question, as well as how much gas was consumed during the operation's execution.
+Users can use the result of this function to filter a transaction receipt's logs into those logs specifically emitted due to the operation, as well as determine the status and gas usage of the operation.
+
+## Backwards Compatibility
+
+This ERC specifies the ideal behaviour of an endorser, but recognizes that some features might not be possible to implement for existing smart contract wallets.
+If it is not possible to infer the gas consumed during an operation's execution, `operationReceipt` MUST return 0 for `gasUsed`.
+If it is not possible to infer exactly those logs that were emitted specifically due to the operation, `operationReceipt` MUST return an empty array for `operationLogs`.
+If it is not possible to infer the success status of the operation, `operationReceipt` MUST return `Status.Unknown` for `status`.
+Finally, if it is not even possible to implement `operationFilter`, `operationFilter` MUST return the zero address and an empty array of topics.
```

Comment:
These are requirements, and so should be in the Specification section.

The Backwards Compatibility section is for discussing the impact of your proposal.